extractor/src/main/java/org/schabi/newpipe/extractor/services/peertube/extractors/PeertubePlaylistExtractor.java

```diff
@@ -1,85 +1,135 @@
 
     @Override
     public String getThumbnailUrl() throws ParsingException {
-        // TODO Auto-generated method stub
-        return null;
+        return getBaseUrl() + playlistInfo.getString("thumbnailPath");
     }
 
     @Override
     public String getBannerUrl() throws ParsingException {
-        // TODO Auto-generated method stub
         return null;
     }
 
     @Override
     public String getUploaderUrl() throws ParsingException {
-        // TODO Auto-generated method stub
-        return null;
+        return playlistInfo.getObject("ownerAccount").getString("url");
     }
 
     @Override
     public String getUploaderName() throws ParsingException {
-        // TODO Auto-generated method stub
-        return null;
+        return playlistInfo.getObject("ownerAccount").getString("displayName");
     }
 
     @Override
     public String getUploaderAvatarUrl() throws ParsingException {
-        // TODO Auto-generated method stub
-        return null;
+        return getBaseUrl() + playlistInfo.getObject("ownerAccount").getObject("avatar").getString("path");
     }
 
     @Override
     public long getStreamCount() throws ParsingException {
-        // TODO Auto-generated method stub
-        return 0;
+        return playlistInfo.getNumber("videosLength").longValue();
+    }
+
+    @Nonnull
+    @Override
+    public String getSubChannelName() throws ParsingException {
+        return playlistInfo.getObject("videoChannel").getString("displayName");
+    }
+
+    @Nonnull
+    @Override
+    public String getSubChannelUrl() throws ParsingException {
+        return playlistInfo.getObject("videoChannel").getString("url");
     }
 
+    @Nonnull
+    @Override
+    public String getSubChannelAvatarUrl() throws ParsingException {
+        return getBaseUrl() + playlistInfo.getObject("videoChannel").getObject("avatar").getString("path");
+    }
+
+    @Nonnull
     @Override
     public InfoItemsPage<StreamInfoItem> getInitialPage() throws IOException, ExtractionException {
-        // TODO Auto-generated method stub
-        return null;
+        return getPage(initialPageUrl);
     }
 
     @Override
     public String getNextPageUrl() throws IOException, ExtractionException {
-        // TODO Auto-generated method stub
-        return null;
+        return PeertubeParsingHelper.getNextPageUrl(initialPageUrl, total);
     }
 
     @Override
     public InfoItemsPage<StreamInfoItem> getPage(String pageUrl) throws IOException, ExtractionException {
-        // TODO Auto-generated method stub
-        return null;
+        Response response = getDownloader().get(pageUrl);
+        try {
+            playlistVideos = JsonParser.object().from(response.responseBody());
+        } catch (JsonParserException jpe) {
+            throw new ExtractionException("Could not parse json", jpe);
+        }
+        PeertubeParsingHelper.validate(playlistVideos);
+
+        this.total = JsonUtils.getNumber(playlistVideos, "total").longValue();
```

Comment:
This shouldn't happen like this, however it's not worth fixing it until my next page PR (where you could just make it a normal variable in the function), especially as it's done like this in the other PeerTube ListExtractors as well.